models/default_local_provider.go

```diff
@@ -491,17 +490,23 @@ func (l *DefaultLocalProvider) RemotePatternFile(req *http.Request, resourceURL,
 	// Check if hostname is github
 	if parsedURL.Host == "github.com" {
 		parsedPath := strings.Split(parsedURL.Path, "/")
+		if parsedPath[3] == "tree" {
```

Comment:
@Revolyssup Meshery server panics for me when the length is shorter than expected here.
I simply used : https://github.com/service-mesh-patterns/service-mesh-patterns
We should maybe place some check?
```
15:53:45 http: panic serving 127.0.0.1:39282: runtime error: index out of range [3] with length 3
goroutine 11692 [running]:
net/http.(*conn).serve.func1()
        /usr/local/go/src/net/http/server.go:1801 +0xb9
panic({0x313f4c0, 0xc0011a4738})
        /usr/local/go/src/runtime/panic.go:1047 +0x266
github.com/layer5io/meshery/models.(*DefaultLocalProvider).RemotePatternFile(0xc000161a40, 0x2d64de0, {0xc0011c0cc0, 0x3e}, {0x0, 0x0}, 0x1)
        /home/rudraksh/dev/meshery/models/default_local_provider.go:493 +0x3e5
github.com/layer5io/meshery/handlers.(*Handler).handlePatternPOST(0x90, {0x367c3e8, 0xc000d1a620}, 0xc00109df00, 0x7f0768af6d28, 0x88, {0x370c3c0, 0xc000161a40})
        /home/rudraksh/dev/meshery/handlers/meshery_pattern_handler.go:124 +0x1106
github.com/layer5io/meshery/handlers.(*Handler).PatternFileRequestHandler(0xc0002c9440, {0x367c3e8, 0xc000d1a620}, 0xc000a1d730, 0x108bb87, 0x90, {0x370c3c0, 0xc000161a40})
        /home/rudraksh/dev/meshery/handlers/meshery_pattern_handler.go:40 +0x6a
github.com/layer5io/meshery/handlers.(*Handler).SessionInjectorMiddleware.func1({0x367c3e8, 0xc000d1a620}, 0xc00109de00)
        /home/rudraksh/dev/meshery/handlers/middlewares.go:118 +0x5cc
net/http.HandlerFunc.ServeHTTP(0xc000d7ad80, {0x367c3e8, 0xc000d1a620}, 0x7f0768af6d28)
        /usr/local/go/src/net/http/server.go:2046 +0x2f
github.com/layer5io/meshery/handlers.(*Handler).AuthMiddleware.func1({0x367c3e8, 0xc000d1a620}, 0xc00109de00)
        /home/rudraksh/dev/meshery/handlers/middlewares.go:65 +0x173
net/http.HandlerFunc.ServeHTTP(0x3690028, {0x367c3e8, 0xc000d1a620}, 0x361aca0)
        /usr/local/go/src/net/http/server.go:2046 +0x2f
github.com/layer5io/meshery/handlers.(*Handler).ProviderMiddleware.func1({0x367c3e8, 0xc000d1a620}, 0xc00109dd00)
        /home/rudraksh/dev/meshery/handlers/middlewares.go:33 +0x316
net/http.HandlerFunc.ServeHTTP(0xc00109dc00, {0x367c3e8, 0xc000d1a620}, 0xc0014219f8)
        /usr/local/go/src/net/http/server.go:2046 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc000c29500, {0x367c3e8, 0xc000d1a620}, 0xc00109db00)
        /home/rudraksh/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
net/http.serverHandler.ServeHTTP({0x366cca8}, {0x367c3e8, 0xc000d1a620}, 0xc00109db00)
        /usr/local/go/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc000ca20a0, {0x3690028, 0xc000d04ba0})
        /usr/local/go/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:3033 +0x4e8
```